### PR TITLE
[Wave Collect][Xero] Enforce 2FA for xero

### PR DESCRIPTION
**Pull request by rushatgabhane**
### Fixed Issues

$ https://github.com/Expensify/App/issues/43015
PROPOSAL:

#### Test 1 - Require 2FA for connection

          Pre-requisites: Have an account that 
          1. does not have 2FA.
          2. is an admin of a control policy.

1. Go to an account that doesn't have 2FA.
2. Go to a collect policy > accounting.
3. Click connect Xero.
4. Verify that 2FA modal is shown.
5. Complete the 2FA steps.
6. After 2FA has been setup, verify that Xero integration page opens.

image

            

#### Test 2 - Require 2FA for admin that signs in

            Pre-requisites: Have an account that 
            1. does not have 2FA.
            2. has Xero accounting connected.
            3. is an admin of the policy.

1. Login using the account.
2. After login, verify that 2FA modal is shown.
3. Complete the 2FA steps.

image

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/95e1ffa1-468a-4623-8e24-68959818c322

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

image

</details>
<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/29673073/124bbe7c-8a07-488c-b799-5b0813e1655d

https://github.com/Expensify/App/assets/29673073/1b131f6c-87f4-48de-8325-23707f23b86b

image

</details>

<details>
<summary>MacOS: Desktop</summary>

image

</details>

### CODE REVIEW

**Comment by rushatgabhane** [comment]
cc @mananjadhav

**Comment by c3024** [comment]
I guess this will be reviewed by @mananjadhav . Is there a way to remove the review request from me myself?

**Comment by rushatgabhane** [comment]
@c3024 since you're online you can review it. I'll help you with things.
Do you wish to review this PR?

**Comment by c3024** [comment]
Yes. Thanks!

**Comment by rushatgabhane** [comment]
@c3024 i've added the test steps

**Comment by c3024** [comment]
Test 1 works well. For Test 2, I guess I need to be added to Xero beta?

**Comment by rushatgabhane** [comment]
@c3024 what is your email? I'll add your account to a Xero Integrated policy

**Comment by c3024** [comment]
chaitanya3024+xero@gmail.com

**Comment by rushatgabhane** [comment]
done, please relogin

File: src/libs/Navigation/types.ts [hunk 1]
```diff
@@ -388,7 +393,10 @@ type SettingsNavigatorParamList = {
     [SCREENS.GET_ASSISTANCE]: {
         backTo: Routes;
     };
-    [SCREENS.SETTINGS.TWO_FACTOR_AUTH]: BackToParams;
+    [SCREENS.SETTINGS.TWO_FACTOR_AUTH]: {
```
Comment (c3024) [inline src/libs/Navigation/types.ts:396]: Can we use `BackToAndForwardToParms` here and in place of `SuccessStepParams` in `SuccessStep`?

Also, in which cases is `forwardTo` applicable?

**Comment by c3024** [comment]
For the Test 2 flow, on logging in the forward button of browser throws a development error

https://github.com/Expensify/App/assets/102477862/91e2a130-8ecf-4729-8419-dd733df415f0

After disabling 2FA again for admins with policies with Xero account connected, sometimes the page hangs. Sometimes right after disabling it and sometimes a little later.

File: src/languages/en.ts [hunk 2]
```diff
@@ -1023,6 +1023,10 @@ export default {
         congrats: 'Congrats, now you’ve got that extra security.',
         copy: 'Copy',
         disable: 'Disable',
+        enableTwoFactorAuth: 'Enable two-factor authentication',
+        pleaseEnableTwoFactorAuth: 'Please enable Two-factor authentication.',
+        twoFactorAuthIsRequiredDescription: 'Two-factor authentication is required for connecting to Xero, please enable Two-factor authentication to continue.',
+        twoFactorAuthIsRequiredForAdminsDescription: 'Two-factor authentication is required for Xero workspace admins, please enable Two-factor authentication to continue.',
```
[the remark this answers is not in the document]
Comment (rushatgabhane) [reply src/languages/en.ts:1029]: @c3024 this is copied straight from figma, so i won't be changing it

File: src/libs/Navigation/types.ts (hunk 1 above)
Comment (rushatgabhane) [reply src/libs/Navigation/types.ts:396]: i don't see any other reuse, so this is fine as is

**Comment by rushatgabhane** [comment]
> After disabling 2FA again for admins with policies with Xero account connected, sometimes the page hangs. Sometimes right after disabling it and sometimes a little later.

@c3024 yes that is raised here. it's unrelated to the PR imo - https://github.com/Expensify/App/issues/43015#issuecomment-2179824461

**Comment by c3024** [comment]
Tests well on Chrome and Desktop.


On iOS, for Test 1, it does not direct to the 2FA page when 2FA is not enabled but goes to Xero login page.



https://github.com/Expensify/App/assets/102477862/046a542d-d725-44a7-8d7c-9ccde58b708a

**Comment by rushatgabhane** [comment]
> On iOS, for Test 1, it does not direct to the 2FA page when 2FA is not enabled but goes to Xero login page.

can you try rebuilding? i cant reproduce the error

**Comment by c3024** [comment]
Okay, it was after a fresh build. Let me try again.

**Comment by rushatgabhane** [comment]
> Pre-requisites: Have an account that 
      1. does not have 2FA.
      2. is an admin of a control policy.

@c3024 it's possible that the account you're testing iOS doesn't meet this pre-requisites

**Comment by c3024** [comment]
I followed [these steps](https://github.com/Expensify/App/pull/40277). That account worked as expected on Chrome and Desktop for Test 1.

**Comment by rushatgabhane** [comment]
@c3024 you're right! I hadn't pushed that code for native.

**Comment by c3024** [comment]
Works well now on iOS native but two small observations.

1. After 2FA addition, Xero connection opened in browser instead of within app
2. Coming back to the app, the accounting and workspace options pages are pushed twice so they appear twice on going back.

https://github.com/Expensify/App/assets/102477862/1120f2ea-b7e8-4002-9d98-91d124d7c488

These two issues do not exist when we visit the Xero connection page again after some other flow.

**Comment by rushatgabhane** [comment]
> Coming back to the app, the accounting and workspace options pages are pushed twice so they appear twice on going back.

Any idea how to fix this?

**Comment by c3024** [comment]
Screenshot 2024-06-20 at 7 24 06 PM

Test2 fails now.

**Comment by rushatgabhane** [comment]
> Test2 fails now

yes that is expected. i was testing test 1, so i had removed xero

File: src/components/ConnectToXeroButton/index.tsx [hunk 3]
```diff
@@ -50,6 +64,17 @@ function ConnectToXeroButton({policyID, shouldDisconnectIntegrationBeforeConnect
                     danger
                 />
             )}
+            {isRequire2FAModalOpen && (
+                <RequireTwoFactorAuthenticationModal
+                    onSubmit={() => {
+                        setIsRequire2FAModalOpen(false);
+                        Navigation.navigate(ROUTES.SETTINGS_2FA.getRoute(ROUTES.POLICY_ACCOUNTING.getRoute(policyID), getXeroSetupLink(policyID)));
```
Comment (c3024) [inline src/components/ConnectToXeroButton/index.tsx:71]:
```suggestion
Navigation.dismissModal();                    Navigation.navigate(ROUTES.SETTINGS_2FA.getRoute(ROUTES.POLICY_ACCOUNTING.getRoute(policyID), getXeroSetupLink(policyID)));
```

I think for narrow layout, another `FullScreenNavigator` is being pushed here because the top navigator is `RightHandModalNavigator`. Dismissing RHP before navigating fixes it. Same for `index.native.tsx`.

Comment (c3024) [reply src/components/ConnectToXeroButton/index.tsx:71]: @rushatgabhane

**Comment by c3024** [comment]
I am having a hard time with building Android locally. 

@francoisl Can you please run a build here?

**Comment by c3024** [comment]
Test 1 tests well on all platforms.

This minor issue is still left.

>After 2FA addition, Xero connection opened in browser instead of within app

Code changes look good as well.

@rushatgabhane let me know when Xero is connected back to the policy again so that I can test Test 2 on remaining platforms.

**Comment by francoisl** [comment]
> let me know when Xero is connected back to the policy again so that I can test Test 2 on remaining platforms.

If it helps, you can simulate a policy being connected to Xero without going through the whole authentication flow:

1. Sign in to any account on [www.expensify.com](http://www.expensify.com/) or [staging.expensify.com](http://staging.expensify.com/)
2. Go to Settings > Workspaces > Create a new workspace
3. Open the workspace
3. In the JS console, run:
```js
const p = Policy.getCurrent();
p.policy.connections = {xero:{config:{isConfigured:true}, data:{}}};
p.save();
```

File: src/languages/en.ts (hunk 2 above)
Comment (francoisl) [inline src/languages/en.ts:1029]: Based on https://expensify.slack.com/archives/C036QM0SLJK/p1718903738453069?thread_ts=1717422173.269869&cid=C036QM0SLJK, let's edit those please:

```suggestion
        pleaseEnableTwoFactorAuth: 'Please enable two-factor authentication.,
        twoFactorAuthIsRequiredDescription: 'Two-factor authentication is required for connecting to Xero. Please enable two-factor authentication to continue.',
        twoFactorAuthIsRequiredForAdminsDescription: 'Two-factor authentication is required for Xero workspace admins. Please enable two-factor authentication to continue.',
```

**Review by francoisl — CHANGES_REQUESTED** [review]

File: src/languages/es.ts [hunk 4]
```diff
@@ -1020,6 +1020,10 @@ export default {
         congrats: 'Felicidades, ahora tienes esa seguridad adicional.',
         copy: 'Copiar',
         disable: 'Deshabilitar',
+        enableTwoFactorAuth: 'Enable two-factor authentication',
+        pleaseEnableTwoFactorAuth: 'Please enable Two-factor authentication.',
+        twoFactorAuthIsRequiredDescription: 'Two-factor authentication is required for connecting to Xero, please enable Two-factor authentication to continue.',
+        twoFactorAuthIsRequiredForAdminsDescription: 'Two-factor authentication is required for Xero workspace admins, please enable Two-factor authentication to continue.',
```
Comment (francoisl) [inline src/languages/es.ts:1026]: We also have the Spanish translations now (https://expensify.slack.com/archives/C036QM0SLJK/p1718881538293029?thread_ts=1717422173.269869&cid=C036QM0SLJK)

```suggestion
        enableTwoFactorAuth: 'Activar la autenticación de dos factores',
        pleaseEnableTwoFactorAuth: 'Activa la autenticación de dos factores.',
        twoFactorAuthIsRequiredDescription: 'La autenticación de dos factores es necesaria para conectarse a Xero. Activa la autenticación de dos factores para continuar.',
        twoFactorAuthIsRequiredForAdminsDescription: 'La autenticación de dos factores es necesaria para los administradores del área de trabajo de Xero. Activa la autenticación de dos factores para continuar.',
```

File: src/components/ConnectToXeroButton/index.tsx (hunk 3 above)
Comment (c3024) [reply src/components/ConnectToXeroButton/index.tsx:71]: Not convinced by this? @rushatgabhane

**Review by c3024 — CHANGES_REQUESTED** [review]

File: src/components/ConnectToXeroButton/index.tsx (hunk 3 above)
Comment (rushatgabhane) [reply src/components/ConnectToXeroButton/index.tsx:71]: @c3024 sorry, let me give this a try

**Comment by rushatgabhane** [comment]
working on this today

**Comment by rushatgabhane** [comment]
@francoisl hey 👋 

Yes that does the job : )

https://github.com/Expensify/App/assets/29673073/d5c527da-e75f-4a99-a973-2ccd5ea578d6

File: src/components/ConnectToXeroButton/index.tsx (hunk 3 above)
Comment (rushatgabhane) [reply src/components/ConnectToXeroButton/index.tsx:67]: this works well!

**Comment by rushatgabhane** [comment]
@c3024 could you please re-review

**Comment by c3024** [comment]
NAB: There is no overlay on Chrome for the modal 

Screenshot 2024-07-07 at 10 04 04 PM

it is fine on Desktop

Screenshot 2024-07-07 at 10 08 21 PM

**Comment by rushatgabhane** [comment]
interestinggg

**Comment by rushatgabhane** [comment]
That looks like a bug with `Modal`. Am I missing any props that I should pass to Modal? I can't find anything.
This is happening after I merged `main`

**Comment by francoisl** [comment]
Yeah I'm also seeing that on `main` anyway, I agree it's unrelated to this PR.
I sent a bug report for it in Slack [here](https://expensify.slack.com/archives/C049HHMV9SM/p1720489736399529), in the meantime we can move forward with this PR.

**Comment by c3024** [comment]
1. Should we disallow disabling 2FA for a Xero configured account?

3. When logging in with a Xero configured but not 2FA enabled account, `Woohoo` screen shows up behind the modal for native apps. This might be out of scope for this issue.

4. NAB - In iOS app, for test 1, after pressing on Xero and enabling 2FA, the `expensify.com.dev` link opens. This should work fine on `staging` and `prod`. What do I need to change so that link opened be of `staging`. It opens `staging` link correctly on Android.

**Comment by francoisl** [comment]
> 1. Should we disallow disabling 2FA for a Xero configured account?

I'd say let's keep this out of scope for this PR, and consider it as a potential improvement for later. Admins will still be prompted to enable 2FA again the next time they sign in, so it's fine.

**Comment by c3024** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by c3024 — APPROVED** [review]
LGTM!

**Comment by c3024** [comment]
Needs lint fix and filling author checklist completion.

**Comment by francoisl** [comment]
@rushatgabhane Looks like there's a small conflict now, can we resolve? Let's try to get this PR merged, we're so close!

**Comment by rushatgabhane** [comment]
@francoisl all resolved and retested. Updated the screenshots with new ones. ✅

**Comment by francoisl** [comment]
Looking good! One issue I'm seeing is that after enabling 2FA, chats don't load at all. Expenses on the Search page initially don't load either, but after switching between tabs they load. Are you guys seeing that on your side too?

https://github.com/user-attachments/assets/34762b53-77ca-4045-abc4-e3c2d317b4b9

**Comment by rushatgabhane** [comment]
@francoisl nice catch! It might be due to bad data. I fixed it by clearing onyx data and then making the api calls.

Here is how it looks - 


https://github.com/user-attachments/assets/5eaf5868-7aa8-491d-a98a-9ae0fcbd16f5

**Comment by francoisl** [comment]
Thanks for the quick fix. Tested again and it works great now 🚀

**Review by francoisl — APPROVED** [review]

**Comment by MonilBhavsar** [comment]
@rushatgabhane @c3024 this might have caused a deploy blocker https://github.com/Expensify/App/issues/45572
Can anyone please take a quick look, thanks!

**Comment by rushatgabhane** [comment]
@MonilBhavsar the issue is present on main too.


https://github.com/Expensify/App/pull/44059/files#diff-2cb1d6b2b2016fb2653f516d359112297412c78fb6a85490c6899d1395226ef6R42-R44

File: src/Expensify.tsx [hunk 5]
```diff
@@ -253,6 +265,16 @@ function Expensify({
                         />
                     ) : null}
                     {focusModeNotification ? <FocusModeNotification /> : null}
+                    {shouldShowRequire2FAModal ? (
+                        <RequireTwoFactorAuthenticationModal
+                            onSubmit={() => {
+                                setShouldShowRequire2FAModal(false);
+                                Navigation.navigate(ROUTES.SETTINGS_2FA.getRoute(ROUTES.HOME));
```
Comment (rojiphil) [inline src/Expensify.tsx:272]: An edge case on ios hybrid required us to call `navigate` only after the modal is closed. This is addressed in #58245
